### PR TITLE
Feature/task service tests

### PR DESCRIPTION
**Pull request by weronika0000**
code refactoring:
-new methods (creating tasks and dtos) that are called in tests
-added "display name" of tests
- "when" changed for "given"

### CODE REVIEW

**Review by sylvvester — APPROVED** [review]
LGTM

File: src/test/java/com/codersfactory/task/TaskServiceImplTest.java [hunk 1]
```diff
@@ -1,160 +1,191 @@
...
+    private static final DifficultyLevel DIFFICULTY_LEVEL = INTERMEDIATE;
+    private static final Long CREATOR_ID = 1L;
+    private static final Instant CREATED_AT= Instant.now();
+    private static final Instant UPDATED_AT= Instant.now();
+    private static final  Duration AVERAGE_COMPLETION_TIME = null;
+    private static final String TECHNOLOGY ="Test technology";
+    private static final String TESTS = "Test tests";
+    private static final boolean IF_APPROVED = false;
+    private static final Set<TaskSolution> TASK_SOLUTION = null;
+    private static final Article ARTICLE_ID = null;
```
Comment (pacion) [inline src/test/java/com/codersfactory/task/TaskServiceImplTest.java:47]: Use more production likely naming, something like

`
TITLE = "Check the task"
`
`
HINT = "It is below the fridge!"
`

File: src/test/java/com/codersfactory/task/TaskServiceImplTest.java [hunk 2]
```diff
@@ -1,160 +1,191 @@
...
     private TaskServiceImpl taskService;
 
+    @DisplayName("test createTask: when parameters are valid should return taskResponseDto")
     @Test
     void testCreateTask_whenValidParametersProvided_thenReturnCreateTaskResponseDto() {
-        //Arrange
-        CreateTaskRequestDto CreateTaskRequestDto = new CreateTaskRequestDto("Test Title", "Test Content", "Test Example Solution", "Test Hint", 5, DifficultyLevel.INTERMEDIATE, 1L, "Test technology", "Test tests");
-        Task taskFromRequest = new Task(null, "Test Title", "Test Content", "Test Example Solution", "Test Hint", 5, DifficultyLevel.INTERMEDIATE, 1L, Instant.now(), Instant.now(), null, "Test technology", "Test tests", false, null, null);
-        Task taskFromDatabase = new Task(1L, "Test Title", "Test Content", "Test Example Solution", "Test Hint", 5, DifficultyLevel.INTERMEDIATE, 1L, Instant.now(), Instant.now(), null, "Test technology", "Test tests", false, null, null);
-        CreateTaskResponseDto taskResponseDto = new CreateTaskResponseDto(1L, "Test Title", "Test Content", "Test Example Solution", "Test Hint", 5, DifficultyLevel.INTERMEDIATE, 1L, Instant.now(), "Test technology", "Test tests");
```
Comment (pacion) [inline src/test/java/com/codersfactory/task/TaskServiceImplTest.java:39]: imo this is way better,
creating object in given section should be in each test, each separated test should contain unique data

File: src/test/java/com/codersfactory/task/TaskServiceImplTest.java [hunk 3]
```diff
@@ -1,160 +1,191 @@
...
+                .hint(HINT)
+                .numberOfPoints(NUMBER_OF_POINTS)
+                .difficultyLevel(INTERMEDIATE)
+                .creatorId(CREATOR_ID)
+                .createdAt(CREATED_AT)
+                .averageCompletionTime(AVERAGE_COMPLETION_TIME)
+                .technology(TECHNOLOGY)
+                .tests(TESTS)
+                .ifApproved(IF_APPROVED)
+                .build();
```
Comment (pacion) [inline src/test/java/com/codersfactory/task/TaskServiceImplTest.java:185]: each task has the same data 🗡️

**Review by pacion — APPROVED** [review]
imo you'd better to focus on writing integrational test instead of mocking everything and unit testing

**Pull request merged**